subsys/bluetooth/host/gatt.c

```diff
@@ -3594,6 +3599,26 @@ void bt_gatt_connected(struct bt_conn *conn)
 	data.conn = conn;
 	data.sec = BT_SECURITY_L1;
 
+#if defined(CONFIG_BT_SETTINGS_CCC_LOAD_ON_DEMAND)
```

Comment:
Can't we use IS_ENABLE in this block?

Reply:
Not without making a certain amount of unrelated changes, it references ccc_set_direct which references ccc_set which is only defined when BT_SETTINGS is on.

I wanted to keep the PR small, but I think it can be done by moving things around (ccc_set can be moved out of the #if defined(CONFIG_BT_SETTINGS)) if you prefer!

Reply:
Addressed in latest push, I separated functionality and moving functions around in separate commits as suggested by @joerchan.